src/transformers/models/modernbert/modular_modernbert.py

```diff
@@ -0,0 +1,1231 @@
+# coding=utf-8
+# Copyright 2024 Google Inc. HuggingFace Inc. team. All rights reserved.
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+from enum import Enum
+from typing import Optional, Tuple, Union
+
+import torch
+import torch.nn as nn
```

Comment:
```suggestion
from torch import nn
```
a nit, but the `as nn` is a bit unnecessary, let's avoid it.